Dockerfile

```diff
@@ -1,11 +1,79 @@
-FROM ubuntu:18.04
+# syntax=docker/dockerfile:1.3-labs
+
+# Build and test image
+# (No need to worry about creating intermediate images)
+
+FROM ubuntu:18.04 AS build
 
 ENV DEBIAN_FRONTEND noninteractive
 
-COPY . /app
+# Install build deps
+RUN <<EOF
+  apt-get update
+  apt-get install --no-install-recommends -y curl python3 python3-pip python3-dev python3-setuptools python3-wheel build-essential git
+EOF
+
 WORKDIR /app
 
-RUN ./build.sh
+# Install Meteor
+COPY .meteor/release /app/.meteor/release
+RUN <<EOF bash
+  set -eux
+  set -o pipefail
+  METEOR_RELEASE="$(sed -e 's/.*@//g' .meteor/release)"
+  curl -sL https://install.meteor.com?release=\$METEOR_RELEASE | sh
+EOF
+
+# Install app deps
+COPY package.json package-lock.json /app
+RUN --mount=type=cache,target=/root/.npm meteor npm ci
+
+# Run lint
+COPY . /app
+RUN meteor npm run lint
+
+# Generate production build
+RUN --mount=type=cache,target=/app/.meteor/local/ meteor build --allow-superuser --directory /built_app --server=http://localhost:3000
+
+# Install server dependencies
+WORKDIR /built_app/bundle/programs/server
+RUN --mount=type=cache,target=/root/.npm meteor npm install --production
+
+# Production image
+# (Be careful about creating as few layers as possible)
+
+FROM ubuntu:18.04
+
+# Install runtime deps
+RUN <<EOF
+  . /etc/os-release
```

Comment:
FTR, I would normally use `lsb_release` here, but it's not installed in the Ubuntu docker container.

Reply:
`/etc/os-release` is pretty darn guaranteed to exist at this point: https://www.freedesktop.org/software/systemd/man/os-release.html

Reply:
Ah good. I was wondering if sourcing the file was a bad idea. Sounds like that's an explicit design goal